features/steps/result_api_steps.py

```diff
@@ -104,19 +116,23 @@ def step_impl(context, name):
                 equal_to(name))
 
 
-@then('the analyser version should be "{version}"')
-def step_impl(context, version):
-    assert_that(context.chosen_result_json['metadata']['analyzer_version'],
-                equal_to(version))
+@then('the analyser version should not be empty')
+def step_impl(context):
+    assert_that(len(context.chosen_result_json['metadata']['analyzer_version']), not equal_to(0))
 
 
-@then(u'the solver should be "{name}"')
+@then('the solver should be "{name}"')
 def step_impl(context, name):
     assert_that(context.chosen_result_json['metadata']['analyzer'],
                 equal_to(name))
 
 
-@then(u'the solver version should be "{version}"')
-def step_impl(context, version):
-    assert_that(context.chosen_result_json['metadata']['analyzer_version'],
-                equal_to(version))
+@then('the solver version should not be empty')
+def step_impl(context):
+    assert_that(len(context.chosen_result_json['metadata']['analyzer_version']), not equal_to(0))
+
+
+@then('the list of analzer results should not be empty')
+def step_impl(context):
+    assert_that(len(context.latest_analyzer_result['metadata']['analyzer']), not equal_to(0))
```

Comment:
This is confusing - the step states "'the list of analzer results should not be empty" but AFAIS it checks that the name of analyzer has non-zero length.

Reply:
@ace2107 ?!